source/common/upstream/health_checker_impl.cc

```diff
@@ -347,9 +360,26 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onGoAway(
 HttpHealthCheckerImpl::HttpActiveHealthCheckSession::HealthCheckResult
 HttpHealthCheckerImpl::HttpActiveHealthCheckSession::healthCheckResult() {
   const uint64_t response_code = Http::Utility::getResponseStatus(*response_headers_);
-  ENVOY_CONN_LOG(debug, "hc response={} health_flags={}", *client_, response_code,
+  ENVOY_CONN_LOG(debug, "hc response_code={} health_flags={}", *client_, response_code,
                  HostUtility::healthFlagsToString(*host_));
 
+  if (!parent_.expected_response_.empty()) {
+    // If the expected response is set, check the first 1024 bytes of actual response if contains
+    // the expected response.
+    std::string response_data =
```

Comment:
Using std::string here causes the data to be copied out of the `response_body_` buffer. Using `absl::string_view` instead avoids this extra copy.

Reply:
Good point, thank you!